packages/apollo-server-lambda/src/ApolloServer.ts

```diff
@@ -213,19 +225,29 @@ export class ApolloServer extends ApolloServerBase {
         }
       }
```

Comment:
I'll make reference to this again in comments below, but is there a reason that most of the logic that's being added in `lambdaApollo.ts` (below, in this PR) isn't added here instead (before `graphqlLambda` is ever kicked off)?

This is more of what I was thinking with my comment on the other PR (https://github.com/apollographql/apollo-server/pull/1739/files#r288265931) and would make this match the other integrations.  This is most important from a maintainability standpoint over the the Apollo Server project as a whole since this divergent pattern makes it difficult to grok what's going on (particularly by starting a new pattern which passes _two_ option objects to `graphqlLambda` which already has an `options` property for its `GraphQLOptions`.

Reply:
I believe the reason that everyone (me and the orig PR) has been organizing the code to process the files first, before doing the actual query is because the `processFileUploads` function's response is actually needed for the query body when it is a multipart form.  We're struggling to follow the same design principles of the other integrations (Express, HAPI, Fastify, etc) because those integrations all have the benefit of actual proper middleware implementations at the server level and that's where those implementations process the uploads. The lambda server/proxy doesn't really have that level of middleware as I understand it. And I'm pretty sure we're not going to get into a complete re-write of that part here. However, in those other integrations, middleware acts much the same way as I understand it, by handling the file upload first and then performing the query.

All that said, I did take another stab at it, moving the logic out to `ApolloServer.ts`. The `graphqlLambda` function kicks off the chain, and we kinda cheat by passing `processEvent` as the first arg so that we can await the file upload processing in `lambdaApollo.ts`. Not sure it really changes anything though, as we still wait for the upload to finish before running the query, but that seems to be the nature of the beast. If you'd like to see this WIP, you can look at the work in this branch: https://github.com/abcya/apollo-server/tree/event-rework/packages/apollo-server-lambda/src
Please ignore the `any` typings all over the place, was just trying to prove a concept quickly. Would definitely need some work there, just didn't want to put in too much effort without discussion (I'm not convinced myself).

The more time we spend on this, the more I see why the original PR structured it the way he did with the `.then()`, and I actually think it's a decent approach given the limitations we're up against.